File: Labs/7-1.c
/*
 * In The Name of God
 * =======================================
 * [] File Name : 7-1.c
 *
 * [] Creation Date : 09-11-2018
 *
 * [] Created By : Parham Alvani <parham.alvani@gmail.com>
 * =======================================
*/
/*
* Copyright (c)  2018 Parham Alvani.
*/
#include <stdio.h>

int main() {
  double d;
  scanf("%lf", &d);

  if ((int) d == d) { // d is an integer
    printf("%lf is an integer\n", d);
  } else {
    printf("%lf isn't an integer (%d)\n", d, (int)d);
  }
}

File: Fall-2019/Labs/Loop/7-1.go
/*
 *
 * In The Name of God
 *
 * +===============================================
 * | Author:        Parham Alvani <parham.alvani@gmail.com>
 * |
 * | Creation Date: 11-11-2019
 * |
 * | File Name:     7-1.go
 * +===============================================
 */

package main

import "fmt"

func main() {
	var n int

	if _, err := fmt.Scanf("%d", &n); err != nil {
		return
	}

	fmt.Println(isUgly(n))
}

func isUgly(num int) bool {
	if num == 0 {
		return false
	}
	for num%2 == 0 {
		num = num / 2
	}
	for num%3 == 0 {
		num = num / 3
	}
	for num%5 == 0 {
		num = num / 5
	}
	return num == 1
}
